### PR TITLE
[Docs] Update API docs (task.base.py)

### PR DESCRIPTION
**Pull request by kotekjedi**
* new sections to api.rst added

* doc in tasks.base.py updated

### CODE REVIEW

File: shifthappens/data/base.py [hunk 1]
```diff
@@ -11,19 +11,21 @@
...
     @abc.abstractmethod
     def __iter__(self):
         raise NotImplementedError
 
 
 class IndexedDataset(Dataset):
     """
-    A class representing a map-style dataset. Your map-style datasets should be
-    inherited from this class.
+    A class representing a map-style dataset (https://pytorch.org/docs/stable/data.html#map-style-datasets).
```
Comment (stes) [inline shifthappens/data/base.py:25]: see above

File: shifthappens/models/base.py [hunk 2]
```diff
@@ -99,7 +99,7 @@ class Model(abc.ABC):
     If your model uses unsupervised adaptation mechanisms override :py:meth:`prepare`
     as well.
 
-    Also make sure that your model is inherited from  at mixins from :py:mod:`shifthappens.model.base`
+    Also make sure that your model is inherited from  at mixins from ``shifthappens.model.base``
```
Comment (stes) [inline shifthappens/models/base.py:102]: this should again use `:py:mod:`, or what is the reason for removing?

File: shifthappens/data/base.py (hunk 1 above)
Comment (zimmerrol) [reply shifthappens/data/base.py]: solved.

File: shifthappens/data/base.py [hunk 3]
```diff
@@ -11,19 +11,21 @@
 
 class Dataset(abc.ABC):
     """
-    An abstract class representing an iterable dataset. Your iterable datasets should be
-    inherited from this class.
+    An abstract class representing an iterable dataset (https://pytorch.org/docs/stable/data.html#iterable-style-datasets).
```
[the remark this answers is not in the document]
Comment (zimmerrol) [reply shifthappens/data/base.py]: solved

File: shifthappens/models/base.py (hunk 2 above)
Comment (zimmerrol) [reply shifthappens/models/base.py]: solved

**Review by zimmerrol — APPROVED** [review]
LGTM

**Review by zimmerrol — APPROVED** [review]
LGTM

**Pull request merged**